src/sentry/static/sentry/app/components/buttonBar.tsx

```diff
@@ -1,9 +1,61 @@
+import React from 'react';
 import styled from '@emotion/styled';
+import space, {ValidSize} from 'app/styles/space';
 
-const ButtonBar = styled('div')`
-  display: flex;
-  justify-content: flex-end;
+type ButtonBarProps = {
+  gap?: ValidSize;
+  merged?: boolean;
+  children: React.ReactNode;
+};
+
+function ButtonBar({children, merged = false, gap = 0}: ButtonBarProps) {
+  return (
+    <ButtonGrid merged={merged} gap={gap} childCount={React.Children.count(children)}>
+      {children}
+    </ButtonGrid>
+  );
+}
+
+const ButtonGrid = styled('div')<{gap: ValidSize; childCount: number; merged: boolean}>`
+  display: grid;
+  grid-template-columns: repeat(${p => p.childCount}, minmax(10px, auto));
```

Comment:
can we use `grid-auto-columns` here?

Reply:
I tried that with the `minmax(10px, auto)` and the grid stacked buttons vertically. I tried a few other values for `grid-auto-columns` without much luck.

Reply:
Oops, what about `grid-auto-flow: column;   grid-template-columns: minmax(10px, auto);`